### ISSUE TITLE
If sdk is not already installed it could install it or provide instructions

### ISSUE THREAD

**Issue by jasperblues** [reitzig/sdkman-for-fish#26]

**Comment by reitzig** [comment]
Could you please elaborate on the use-case you have in mind? Why would one install _this_ before SDKMAN! itself?

**Comment by jasperblues** [comment]
Brand new mac. Installed fish shell. Installed fisher. Needed a Java tool, so installed sdkman-for-fish - expected it to 'just work', however it didn't. Took a moment to figure out why. I don't think it was unreasonable to expect, like a package manager, that the tool would install sdkman itself as a 'transitive dependency', though of course it makes sense why it didn't.

**Comment by reitzig** [comment]
As confusing (and flattering) it is to me that one would find a mere helper script before the real thing, I see your point.

I'll see what I can do. Since installation is as easy as running

	curl -s "https://get.sdkman.io" | bash

it seems very feasible to install SDKMAN! if missing (after user confirmation).

**Comment by jasperblues** [comment]
Nice!

**Comment by tonsV2** [comment]
Google brought mere. I didn't have sdkman installed and it also took me some time to figure out that I needed to install it myself. And also that I had to rerun "fisher add reitzig/sdkman-for-fish" again afterwards.

**Comment by reitzig** [comment]
I changed the language in README to manage expectation.

I also added a simple call to the SDKMAN! installer, so to speak, to function `sdk` that can be run if SDKMAN! is not yet installed.

> And also that I had to rerun "fisher add reitzig/sdkman-for-fish" again afterwards.

That should never be necessary. You _do_ have to open a new shell, though.
(I tried `source`-ing the `sdk.fish` files after installation of SDKMAN! but that didn't work. PRs welcome.)
